drafts/pair-programming-mistakes.md

```diff
@@ -0,0 +1,66 @@
+
+Pair programming is an excellent opportunity for knowledge sharing, but it’s crucial that this exchange happens in digestible bites, not all at once. This is especially important when pairing with someone who is eager to improve and prove themselves. In this particular example, I should have told him the IDE shortcuts at the end of our pairing session. Since then, I learned to prioritize establishing a trusting relationship where it’s okay to fail and not know everything. Additionally, aside from building a trusting dynamic, it's important to establish an environment where asking questions is encouraged. By asking questions, your partner can better understand where you are in your thought process. Sometimes, you have assumptions about what your partner knows or what they are thinking, and asking questions is perfect for quickly getting on the same page. This approach is especially beneficial when working with someone new to the team, as it helps alleviate their nervousness and allows them to focus more on learning and problem-solving.
+
+
+## The Trap of No Breaks
+Despite being aware of the importance of breaks and switch-ups, I often find myself falling into the trap of continuous pairing. When [I’m in the flow](https://en.wikipedia.org/wiki/Flow%5F(psychology)#:~:text=The%20flow%20state,abilities.%22%5B4%5D), time and the need for rest seems to slip away unnoticed, which also happens often when pairing. The driver does not switch with the navigator after a committable segment, and we continue to push forward. Contrary to popular belief, [being in the flow isn’t always advantageous](https://en.wikipedia.org/wiki/Flow%5F(psychology)#:~:text=In%20some%20cases,on%20attentional%20abilities.), since it can lead to fatigue and overworking. I noticed that when we are tired, the quality and quantity of our communication rapidly decreases. We all start to think about our own problem and ignore our partner, or we are stuck on the wrong problem, a phenomenon known as the [XY problem](https://en.wikipedia.org/wiki/XY%5Fproblem). This significantly reduces our productivity and may even render us ineffective.
+
+![Tired of working non-stop.](https://cdn.hashnode.com/res/hashnode/image/upload/v1717162596435/FRNbO6MkS.png?auto=format)
+
+The key in avoiding this trap is to remember to take regular breaks, switch roles and ensure effective communication at all times. This not only helps to keep our minds fresh but also ensures that we are on the same page throughout the whole session.
+
+
+## Remote Pairing under Non-Ideal Conditions
+There was an instance when a colleague and I attempted to pair program remotely. We experienced initially minor, but with time major delay, which caused us frustration and irritation. However, not wanting to constantly complain, we tried to cope with the situation.
```

Comment:
We experienced initially minor, but with time major delay, which caused us frustration and irritation. 

We initially experienced minor delays, but over time they became major, causing us significant frustration and irritation.